src/lib.rs

```diff
@@ -8,6 +8,7 @@ mod levels;
 mod memtable;
 mod ops;
 mod opt;
+mod structs;
```

Comment:
What does `structs` mean?

Reply:
structure-s? What about moving `AgateIterator` in `structs` to a separate mod?

Reply:
Perhaps give it a more meaningful name. Memtable, Skiplist are structs too.

Reply:
I have renamed that to `iterator_trait`.